cmake/OpenCVDetectWebNN.cmake

```diff
@@ -0,0 +1,22 @@
+ocv_clear_vars(HAVE_WEBNN)
+ocv_clear_vars(WEBNN_EMSDK)
+if(WITH_WEBNN)
+  set(WEBNN_HEADER_DIRS "$ENV{WEBNN_NATIVE_DIR}/out/Release/gen/src/include")
+  set(WEBNN_INCLUDE_DIRS "$ENV{WEBNN_NATIVE_DIR}/src/include")
+  set(WEBNN_LIBRARIES "$ENV{WEBNN_NATIVE_DIR}/out/Release/libwebnn_native.so")
+endif()
+
+try_compile(VALID_WEBNN
+      "${OpenCV_BINARY_DIR}"
+      "${OpenCV_SOURCE_DIR}/cmake/checks/webnn.cpp"
```

Comment:
And I think you should also compile the `$ENV{WEBNN_NATIVE_DIR}/gen/src/webnn/webnn_cpp.cpp`

Reply:
fixed